### ISSUE TITLE
Can not deduce `reduce_axis` for ReduceMax: only one in_port and no `axis` parameter.

### ISSUE THREAD

**Issue by nehasoni3** [openvinotoolkit/openvino#256]
Hi
I am getting this error while converting ONNX model into IR files using OpenVino 2019.2.242,
Model Optimizer version: 2019.2.0-436-gf5827d4

I have used this command to generate IR files.
`python3 mo.py --input_model /path/ABC.onnx --output_dir /path/ -b 1 --disable_fusing`

and, following error 
>  [ ERROR ]  Exception occurred during running replacer "None" (<class 'extensions.front.reduce_axis_normalizer.ReduceAxisNormalizer'>): Can not deduce `reduce_axis` for ReduceMax: only one in_port and no `axis` parameter.

Please help me to get out of this.

**Comment by shubha-ramani** [comment]
Dear @nehasoni3 

As I've told you before, I can't debug what's wrong based on an error message alone. if you could zip up your model and attach it to this ticket, that would be great - I could then debug the error.

The error is exactly coming from this code :
C:\Program Files (x86)\IntelSWTools\openvino_2019.2.275\deployment_tools\model_optimizer\extensions\front\reduce_axis_normalizer.py

The node which is 'ReduceMax' doesn't have an 'axis' parameter.

Hope it helps,

Thanks,

Shubha

**Comment by lazarevevgeny** [comment]
@nehasoni3 is this issue still actual?

**Comment by rsrrohit** [comment]
I am still getting the same error. I have attached the onnx file and the config file with this comment as well.

`[setupvars.sh] OpenVINO environment initialized
/content/openvino_training_extensions/external/mmdetection/mmdet/models/detectors/base.py:147: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  imgs_per_gpu = int(imgs[0].size(0))
/content/openvino_training_extensions/external/mmdetection/mmdet/models/backbones/imgclsmob.py:42: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  s = str(i) + ' ' + str(y.shape)
/content/openvino_training_extensions/external/mmdetection/mmdet/models/anchor_heads/anchor_head.py:304: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert cls_score.size()[-2:] == bbox_pred.size()[-2:]
/content/openvino_training_extensions/external/mmdetection/mmdet/core/bbox/transforms.py:51: TracerWarning: torch.as_tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  min_val = torch.as_tensor(min, dtype=dtype, device=device)
/content/openvino_training_extensions/external/mmdetection/mmdet/core/bbox/transforms.py:56: TracerWarning: torch.as_tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  max_val = torch.as_tensor(max, dtype=dtype, device=device)
/content/openvino_training_extensions/external/mmdetection/mmdet/core/post_processing/bbox_nms.py:64: TracerWarning: Converting a tensor to a Python index might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  for i in range(num_classes):
/content/openvino_training_extensions/external/mmdetection/mmdet/core/post_processing/bbox_nms.py:66: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if not cls_inds.any():
/content/openvino_training_extensions/external/mmdetection/mmdet/core/post_processing/bbox_nms.py:69: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if multi_bboxes.shape[1] == 4:
/content/openvino_training_extensions/external/mmdetection/mmdet/ops/nms/nms_wrapper.py:50: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if dets_th.shape[0] == 0:
/content/openvino_training_extensions/external/mmdetection/mmdet/core/post_processing/bbox_nms.py:85: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if bboxes.shape[0] > max_num:
/content/openvino_training_extensions/external/mmdetection/mmdet/core/utils/misc.py:95: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  k = torch.tensor([k], dtype=torch.long)
/content/openvino_training_extensions/external/mmdetection/mmdet/core/utils/misc.py:105: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  values, keep = torch.topk(x, n, dim=dim, **kwargs)
ONNX model has been saved to "/content/config.onnx"
mo.py --input_model="/content/config.onnx" --mean_values="[0, 0, 0]" --scale_values="[255, 255, 255]" --output_dir="/content/" --output="boxes,labels" --input_shape="[1, 3, 256, 256]" --reverse_input_channels
Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	/content/config.onnx
	- Path for generated IR: 	/content/
	- IR output name: 	config
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	boxes,labels
	- Input shapes: 	[1, 3, 256, 256]
	- Mean values: 	[0, 0, 0]
	- Scale values: 	[255, 255, 255]
	- Scale factor: 	Not specified
	- Precision of IR: 	FP32
	- Enable fusing: 	True
	- Enable grouped convolutions fusing: 	True
	- Move mean values to preprocess section: 	False
	- Reverse input channels: 	True
ONNX specific parameters:
Model Optimizer version: 	2019.3.0-408-gac8584cb7
[ ERROR ]  Can not deduce `reduce_axis` for ReduceMax: only one in_port and no `axis` parameter.
Exception occurred during running replacer "None" (<class 'extensions.front.reduce_axis_normalizer.ReduceAxisNormalizer'>): Can not deduce `reduce_axis` for ReduceMax: only one in_port and no `axis` parameter.`

[https://drive.google.com/file/d/1O6FOXp8y_oqJKT0SdcfC6QdfDHFpDoAa/view?usp=sharing](url)
[https://drive.google.com/file/d/1qV50jJwILlyh2yssRhUbiZb5SYRZ6I4v/view?usp=sharing](url)

**Comment by rsrrohit** [comment]
Never mind. I solved the issue by installing the latest Openvino distribution **2020.3.194**. I was using 2019.3.xxx which caused the problem.

**Comment by lazarevevgeny** [comment]
Great to hear that the model works now! Closing the ticket.
